scripts/llnl/common_build_functions.py

```diff
@@ -309,11 +309,14 @@ def set_group_and_perms(directory):
     return 0
 
 
-def full_build_and_test_of_tpls(builds_dir, timestamp):
+def full_build_and_test_of_tpls(builds_dir, timestamp, spec):
     project_file = "scripts/uberenv/project.json"
     config_dir = "scripts/uberenv/spack_configs/{0}".format(get_system_type())
 
-    specs = get_specs_for_current_machine()
+    if spec:
+        specs = [spec]
```

Comment:
i forget to prefix with % all the time.. can you add this:

```
        if not spec.startswith("%"):
            spec = "%{0}".format(spec)
```

Reply:
I had intentionally removed it because the specs used in the CI script (`SPEC: "@develop%${COMPILER}"`) are more than just the compiler and would be changed incorrectly to `%@develop%${COMPILER}`

Reply:
ah.  though i don't know when we wouldnt want develop.  But thats fine.

Reply:
to answer my own question.. when we have a main branch.  need sleep.